packages/flutter/lib/src/services/asset_bundle.dart

```diff
@@ -212,11 +213,23 @@ abstract class CachingAssetBundle extends AssetBundle {
 
 /// An [AssetBundle] that loads resources using platform messages.
 class PlatformAssetBundle extends CachingAssetBundle {
+
+  /// Creates a [PlatformAssetBundle] with the specified [binaryMessenger].
+  ///
+  /// A default [BinaryMessenger] instance will be used if [binaryMessenger] is
+  /// null.
+  PlatformAssetBundle({this.binaryMessenger}) {
+    binaryMessenger ??= ServicesBinding.instance.provideBinaryMessenger();
```

Comment:
this can be an initializer, which allows the field to be final.

that said, see later comments for an alternative strategy that would be consistent with what we'll have to do below.

Reply:
Done.